salt/states/file.py

```diff
@@ -7961,8 +8006,6 @@ def serialize(
         serializer_options["json.serialize"].update({"ensure_ascii": False})
 
     ret = {"changes": {}, "comment": "", "name": name, "result": True}
-    if not name:
-        return _error(ret, "Must provide name to file.serialize")
```

Comment:
Why is this piece being removed?

Reply:
A lapse on my part - thought it was trying to check missing not empty, which obviously doesn't make sense. It'll be restored if the premise of the PR is accepted .